libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -65,7 +66,10 @@ func newKubernetesAnnotator(cfg *common.Config) (processors.Processor, error) {
 		Indexing.RUnlock()
 	}
 
-	metaGen := kubernetes.NewMetaGenerator(config.IncludeAnnotations, config.IncludeLabels, config.ExcludeLabels)
+	//Checks whether the PodUIDIndexer has been added in order to turn on the pod uid gathering
+	_, hasPodUIDIndexer := Indexing.indexers["pod_uid"]
```

Comment:
I think it's better if we don't do this automatically, the user must enable it. In any case, the indexer would still work without this, isn't it?

Reply:
Hmmm the intention here was to enable the gathering of `pod.uid` if the indexer was specified in the config, even if the config bool `include_pod_uid` was not.
I thought this is what you meant with your previous comment: `The indexer would pass the right parameter (enable pod.uid) without the user asking for it.`
Or perhaps I am missing something...

Reply:
Hi,

Yes, I apologize as that is what I said, then I realized the indexer doesn't need the pod uid in the metadata :innocent: 

Still, I think having `pod.uid` is a good feature for some people, so I would keep it as something optional you can enable. I would only remove this auto-enable thingy and we are ready to go.

Reply:
Awesome!
I'll remove the auto-enable and add the info in the changelog first thing tomorrow.
Thanks man! And no need to apologise :slightly_smiling_face: